src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -360,8 +345,7 @@ private async Task ForceAuthenticationAsync<TIOAdapter>(TIOAdapter adapter, bool
 
                 if (!handshakeCompleted)
                 {
-                    // get ready to receive first frame
-                    _handshakeBuffer = new ArrayBuffer(InitialHandshakeBufferSize);
+                    _buffer = new SslBuffer(ReadBufferSize);
```

Comment:
Since `SslBuffer` is struct, I'm wondering if we can just initialized automatically in constructor and here simply have function to allocate/resize/validate the actual buffer. -> EnsureAvailableSpace()

Reply:
Done